### PR TITLE
Add command to query contract information by address.

### PR DESCRIPTION
**Pull request by atchapcyp**
This PR is to implement loomchain command to query contract information by contract address.
for example `loom contract default:0x1234...`
In order to do that we need...

- Add rpc client in go-loom that call `contractrecord` method to loomchain
- Add `contractrecord` route on query service
- Add function GetContractRecord on query server and middleware
- Add `loom contract` command
- Add unit test and E2E test

ref : https://github.com/loomnetwork/go-loom/pull/428, https://github.com/loomnetwork/loomchain/issues/450

- [ ] I added unit tests for any code that added
- [ ] I updated the CHANGELOG.md 
- [ ] All IP is original and not copied from another source
- [ ] I assign all copyright to Loom Network for the code in the pull request

### CODE REVIEW

File: rpc/query_server.go [hunk 1]
```diff
@@ -578,6 +578,27 @@ func (s *QueryServer) ContractEvents(
...
+	if err != nil {
+		return nil, err
+	}
+	snapshot := s.StateProvider.ReadOnlyState()
+	defer snapshot.Release()
+
+	reg := s.CreateRegistry(snapshot)
+	rec, err := reg.GetRecord(contractAddr)
+	if err != nil {
+		return nil, err
```
Comment (enlight) [inline rpc/query_server.go]: -> `return nil, errors.Wrapf(err, "no contract exists at %s", contractAddr.String())`

File: rpc/query_server.go [hunk 2]
```diff
@@ -1136,7 +1157,7 @@ func getReceiptByTendermintHash(state loomchain.State, blockStore store.BlockSto
 	}
 	txHash, err := eth.DecDataToBytes(txObj.Hash)
 	if err != nil {
-		return eth.TxObjToReceipt(txObj, contractAddr), errors.Wrapf(err, "invalid loom transaction hash %h", txObj.Hash)
+		return eth.TxObjToReceipt(txObj, contractAddr), errors.Wrapf(err, "invalid loom transaction hash %v", txObj.Hash)
```
Comment (enlight) [inline rpc/query_server.go]: Is this a merge error?

File: rpc/query_server_test.go [hunk 3]
```diff
@@ -441,3 +442,32 @@ func testQueryServerContractEventsNoEventStore(t *testing.T) {
 		require.NotNil(t, err)
 	})
 }
+
+func testQueryServerGetContractRecord(t *testing.T) {
+
```
Comment (enlight) [inline rpc/query_server_test.go]: Get rid of the blank line

File: rpc/query_server_test.go [hunk 4]
```diff
@@ -441,3 +442,32 @@ func testQueryServerContractEventsNoEventStore(t *testing.T) {
...
+	handler := MakeQueryServiceHandler(qs, testlog, bus)
+	ts := httptest.NewServer(handler)
+	defer ts.Close()
+	// give the server some time to spin up
+	time.Sleep(100 * time.Millisecond)
+
+	rpcClient := rpcclient.NewJSONRPCClient(ts.URL)
+
+	t.Run("Contract query should return error", func(t *testing.T) {
+
```
Comment (enlight) [inline rpc/query_server_test.go]: Get rid of the blank line

File: rpc/query_server.go [hunk 5]
```diff
@@ -578,6 +578,27 @@ func (s *QueryServer) ContractEvents(
...
+
+	reg := s.CreateRegistry(snapshot)
+	rec, err := reg.GetRecord(contractAddr)
+	if err != nil {
+		return nil, err
+	}
+	k := &types.ContractRecordResponse{
+		ContractName:    rec.GetName(),
+		ContractAddress: rec.GetAddress(),
+		CreatorAddress:  rec.GetOwner(),
```
Comment (enlight) [inline rpc/query_server.go]: You can just use `rec.Name` etc. here, no need to use the getters. The getters only deal with the case where `rec == nil`, but that won't happen here.

File: e2e/engine/cmd.go [hunk 6]
```diff
@@ -151,6 +151,12 @@ func (e *engineCmd) Run(ctx context.Context, eventC chan *node.Event) error {
 					}
 					fmt.Printf("--> output:\n%s\n", out)
 
+					if n.SaveContractAddress {
+						outStr := string(out)
+						index := strings.Index(outStr, "default:")
+						e.conf.ContractAddressList = append(e.conf.ContractAddressList, outStr[index:index+50])
```
Comment (enlight) [inline e2e/engine/cmd.go]: This is too hacky, better to just query a contract by name.

File: cmd/loom/loom.go [hunk 7]
```diff
@@ -437,6 +438,51 @@ func newRunCommand() *cobra.Command {
...
+
+func contractInfoCommand() *cobra.Command {
+	var flags cli.ContractCallFlags
+	cmd := &cobra.Command{
+		Use:     "contract [ChainID:Address]",
+		Short:   "Get contract information by address",
+		Args:    cobra.MinimumNArgs(1),
+		Example: contractInfoCommandExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			addr, err := cli.ParseAddress(args[0], flags.ChainID)
```
Comment (enlight) [inline cmd/loom/loom.go]: Use `cli.ResolveAddress` so that it's possible to pass in a contract name or a contract address.

File: Makefile [hunk 8]
```diff
@@ -28,7 +28,7 @@ BINANCE_TGORACLE_DIR=$(GOPATH)/src/$(PKG_BINANCE_TGORACLE)
 # NOTE: To build on Jenkins using a custom go-loom branch update the `deps` target below to checkout
 #       that branch, you only need to update GO_LOOM_GIT_REV if you wish to lock the build to a
 #       specific commit.
-GO_LOOM_GIT_REV = HEAD
+GO_LOOM_GIT_REV = query-contract-record
```
Comment (enlight) [inline Makefile]: This branch has been merged in, can change this back to HEAD

File: rpc/query_server.go (hunk 2 above)
Comment (atchapcyp) [reply rpc/query_server.go]: Whenever I run `rpc/query_server_test.go` I always got the error `Wrapf format %h has unknown verb h` and test build failed 
So should I remain using %h or change it to %X instead?

Comment (enlight) [reply rpc/query_server.go]: It should be `%x` instead of `%h`

**Pull request merged**